### PR TITLE
Use Metadata class from microdata-validator

### PR DESCRIPTION
**Pull request by pawbu**

### CODE REVIEW

**Review by vaskalan — APPROVED** [review]
Nice!

File: datastore_version_manager/domain/draft_dataset.py [hunk 1]
```diff
@@ -57,24 +59,26 @@ def add_new_draft_dataset(operation_type: str,
...
+def _patch_metadata(built_metadata_path: str, dataset_name: str):
     latest_version = datastore.get_latest_version()
     latest_metadata = datastore.get_metadata(
         dataset_name, latest_version
     )
     built_metadata = built_datasets.get_metadata(dataset_name)
-    patched = metadata_patcher.patch(built_metadata, latest_metadata)
+    latest = Metadata(latest_metadata)
+    latest.patch(Metadata(built_metadata))
+    patched = latest.to_dict()
```
Comment (DanielElisenberg) [inline datastore_version_manager/domain/draft_dataset.py:77]: Unless I'm missing this somewhere else: This also needs to update the metadata_all__DRAFT.json

File: tests/unit/domain/test_draft_dataset.py [hunk 2]
```diff
@@ -135,6 +136,30 @@ def test_add_new_draft_dataset_not_built():
...
+    draft_metadata_all = datastore.get_metadata_all('DRAFT')
+
+    assert {
+               "name": "TEST_DATASET_3",
+               "operation": "PATCH_METADATA",
+               "description": "Some fields need patching",
+               "releaseStatus": "DRAFT"
+           } in pending_operations["dataStructureUpdates"]
+
+    test_dataset_3 = [
```
Comment (DanielElisenberg) [inline tests/unit/domain/test_draft_dataset.py:154]: disregard the last comment I've definitely missed it 😅

**Review by DanielElisenberg — APPROVED** [review]
:100:

**Pull request merged**